### ISSUE TITLE
There should be a possibility to set the server should fail to start when modrinth mods cannot get resolved

### ISSUE THREAD

**Issue by michiruf** [itzg/docker-minecraft-server#1903]
### Enhancement Type

Improve an existing feature

### Describe the enhancement

Since it is possible, that modrinth mods can be taken down (either by the owner, or the modrinth team), it should be possible to specify an environment variable to stop the server from starting if mod resolution using modrinth did fail for any mod.

If a server is run with an essential mod, that would otherwise cause data inconsistencies if not installed, this could cause alot of trouble and not running the server might be the solution. In my case this was an inventory synchronizing mod, that has not the possibility to merge data on startup and not starting the server would have helped.
I know that there is the `REMOVE_OLD_MODS` flag, which would also prevent this behaviour, but having a clean setup while trying to set up the server is kind of essential to me, so I always have this setting set to true.

A suggestion for the env could be `MODRINTH_FAIL_ON_MOD_NOT_FOUND` which defaults to the current behaviour `false`.

**Comment by itzg** [comment]
Oh, this might be a bug then. My intention was to fail-fast if any mod doesn't resolve. Can you point me to an example of this? Or is it hypothetical at this point?

**Comment by michiruf** [comment]
Yeah youre right, if the project does not exist, the server will not start. I ment the case, that the mod does not exist in the given scope of the `MODRINTH_ALLOWED_VERSION_TYPE`, so there is no mod being downloaded from modrinth, altho the mod project exists on modrinth. For example:

```yaml
    VERSION: 1.19.2
    MODRINTH_PROJECTS: fabricproxy-lite
    MODRINTH_ALLOWED_VERSION_TYPE: release
```

will result in these mods:
```
test    | [17:19:39] [main/INFO]: Loading 3 mods:
test    |       - fabricloader 0.14.12
test    |       - java 17
test    |       - minecraft 1.19.2
```

(FabricProxy-Lite as of today is only available in beta for 1.19.2, see https://modrinth.com/mod/fabricproxy-lite)

**Comment by itzg** [comment]
Thanks for the details. Yeah, I'm still thinking it's a bug that it doesn't fail when doing that final resolution.

**Comment by itzg** [comment]
This should now be fixed by #2128
